### PR TITLE
Replace Solidity Contract With Vyper

### PR DESCRIPTION
**Pull request by nisdas**
This is a part of #1318. The Solidity Contract will be deprecated and replaced by the vyper contract.

- [x] Add in bindings for the vyper contract
- [x] Ensure any references to the old sol contract are replaced.

### CODE REVIEW

File: contracts/deposit-contract/bytecode.json [hunk 1]
```diff
@@ -0,0 +1 @@
...111561084657600080fd5b6020806121e0826020602088068803016000612060516020826121600101526020810190508061216052612160905001600060046015f15050818152809050905090508051602001806120c0828460006004600a8704601201f16108a957600080fd5b50506000600160e05260c052604060c020546122a052604061226052612260516122c0526120c0805160200180612260516122a001828460006004600a8704601201f16108f557600080fd5b5050612260516122a00161224081516020818352015b60206122405110151561091d5761093a565b6000612240516020850101535b815160010180835281141561090b575b5050506020612260516122a0015160206001820306601f8201039050612260510101612260527fd1faa3f9bca1d698df559716fe6d1c9999155b38d3158fffbc98d76d568091fc612260516122a0a15b5b005b63c5f2892f60005114156109c05734156109a657600080fd5b6000600160e05260c052604060c0205460005260206000f3005b63118e45756000511415610a7657602060046101403734156109e157600080fd5b61014051640100000000610140510110156109fb57600080fd5b64010000000061014051016105605261058060006020818352015b60006001610560511860e05260c052604060c020546101606105805160208110610a3f57600080fd5b60200201526105606002610a5257600080fd5b60028151048152505b8151600101808352811415610a16575b5050610400610160f3005b60006000fd5b610004610a8003610004600039610004610a80036000f3
```
Comment (prestonvanloon) [inline contracts/deposit-contract/bytecode.json]: This isnt valid json

Comment (prestonvanloon) [reply contracts/deposit-contract/bytecode.json]: Maybe call it `bytecode.bin`, i'm not sure what the preferred extension is

File: contracts/deposit-contract/depositContract.v.py [hunk 2]
```diff
@@ -1,20 +1,24 @@
 ## compiled with v0.1.0-beta.7 ##
 
-MIN_DEPOSIT: constant(uint256(wei)) = as_wei_value(1, "ether")
-MAX_DEPOSIT: constant(uint256(wei)) = as_wei_value(32, "ether")
```
Comment (prestonvanloon) [inline contracts/deposit-contract/depositContract.v.py]: Can you revert this change? it is easier to understand 32 ether than it is to understand the gwei amounts

Comment (nisdas) [reply contracts/deposit-contract/depositContract.v.py]: This is actually part of the spec, where gwei is to be used as the primary unit of account. Referenced here: https://github.com/ethereum/eth2.0-specs/pull/460

Comment (prestonvanloon) [reply contracts/deposit-contract/depositContract.v.py]: Right, but you are using `as_wei_value` with the unit specified. So this change just adds unnecessary cognitive friction. 

 `as_wei_value(32, "ether") == as_wei_value(32000000000, "gwei")` so why change it?

File: contracts/deposit-contract/depositContract.v.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
+## compiled with v0.1.0-beta.7 ##
+
+MIN_DEPOSIT: constant(uint256) = 1000000000 # In terms of gwei
+MAX_DEPOSIT: constant(uint256) = 32000000000 # In terms of gwei
```
Comment (prestonvanloon) [inline contracts/deposit-contract/depositContract.v.py]: Ah, I see this now

File: contracts/deposit-contract/depositContract.v.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+Deposit: event({previous_deposit_root: bytes32, data: bytes[2064], merkle_tree_index: bytes[8]})
+ChainStart: event({deposit_root: bytes32, time: bytes[8]})
+
+CHAIN_START_FULL_DEPOSIT_THRESHOLD: uint256
+deposit_tree: map(uint256, bytes32)
+deposit_count: uint256
+full_deposit_count: uint256
+
+@public
+def __init__(depositThreshold: uint256):
```
Comment (prestonvanloon) [inline contracts/deposit-contract/depositContract.v.py]: Can we also set the deposit size? 

For example, in the testnet, I would like to set this to 32gwei rather than 32 ether so that it isn't so "expensive" to test validator deposits.

Comment (nisdas) [reply contracts/deposit-contract/depositContract.v.py]: Alright set in a deposit size argument

File: beacon-chain/powchain/service_test.go [hunk 5]
```diff
@@ -423,7 +419,7 @@ func TestProcessDepositLog(t *testing.T) {
 
 	web3Service.depositTrie = trie.NewDepositTrie()
 
-	currentRoot := web3Service.depositTrie.Root()
+	//currentRoot := web3Service.depositTrie.Root()
```
Comment (rauljordan) [inline beacon-chain/powchain/service_test.go]: why is this here? remove

File: beacon-chain/powchain/service_test.go [hunk 6]
```diff
@@ -635,7 +629,14 @@ func TestProcessChainStartLog(t *testing.T) {
 		t.Fatalf("Unable to retrieve logs %v", err)
 	}
 
-	logs[len(logs)-1].Topics[1] = currentRoot
+	for i := 0; i < 8; i++ {
```
Comment (rauljordan) [inline beacon-chain/powchain/service_test.go]: Magic number, prefer abstract into a constant or put it in config?

File: beacon-chain/powchain/service.go [hunk 7]
```diff
@@ -258,14 +250,15 @@ func (w *Web3Service) ProcessDepositLog(VRClog gethTypes.Log) {
 // ProcessChainStartLog processes the log which had been received from
 // the POW chain by trying to determine when to start the beacon chain.
```
Comment (terenc3t) [inline beacon-chain/powchain/service.go]: Let's to a replace all for `POW` -> `ETH1.0` to align with the spec

File: contracts/deposit-contract/POWlogs.go [hunk 8]
```diff
@@ -0,0 +1,46 @@
+package depositcontract
```
Comment (terenc3t) [inline contracts/deposit-contract/POWlogs.go]: I prefer `ETH1Logs.go` than `POWLogs.go` for clarify

File: contracts/validator-registration-contract/README.md [hunk 9]
```diff
@@ -1,4 +1,4 @@
-## Validator Registration Contract
+## Validator Registration Contract(Deprecated)
```
Comment (terenc3t) [inline contracts/validator-registration-contract/README.md]: feel free to delete this file as well

File: shared/mathutil/math_helper.go [hunk 10]
```diff
@@ -29,3 +29,13 @@ func CeilDiv8(n int) int {
 func IsPowerOf2(n uint64) bool {
 	return (n & (n - 1)) == 0
 }
+
+// PowerOf2 returns an integer that is the provided
```
Comment (terenc3t) [inline shared/mathutil/math_helper.go]: Thanks for this! I'm going to use it for epoch processing!
